src/data/datasets/representation/validation.py

```diff
@@ -0,0 +1,288 @@
+            img_list_csv_path: path to mapping image identifiers to image paths. Format: id | path.
+                Id from matches csv are linked to id from img_list csv
+            transform: Transform to be applied on a sample. This should have the
+                interface of transforms in `albumentations` library.
+            augment: Optional augment to be applied on a sample.
+                This should have the interface of transforms in `albumentations` library.
+            use_gallery: If True, will use gallery data(non-relevant items)
+            gallery_folder: path to a folder with all gallery images (traversed recursively)
+            gallery_list_csv_path: path to mapping image identifiers to image paths. Format: id | path.
+            image_dtype: Data type of of the torch tensors related to the image.
+            img_list_map_column: Image maping column names. Key - TorchOK column name, Value - csv column name.
+            matches_map_column: Matches maping column names. Key - TorchOK column name, Value - csv column name.
+            gallery_map_column: Gallery maping column names. Key - TorchOK column name, Value - csv column name.
+            grayscale: If True, image will be read as grayscale otherwise as RGB.
+
+        Raises:
+            ValueError: if use_gallery True, but gallery_folder or gallery_list_csv_path is None
+        """
+        super().__init__(data_folder, transform, augment, image_dtype, grayscale)
+        self.__matches_map_column = matches_map_column
+        self.__img_list_map_column = img_list_map_column
+        self.__gallery_map_column = gallery_map_column
+
+        self.__matches = pd.read_csv(self._data_folder / matches_csv_path,
+                                     usecols=[self.__matches_map_column['query'], 
+                                              self.__matches_map_column['relevant'],
+                                              self.__matches_map_column['scores']],
+                                     dtype={self.__matches_map_column['query']: int, 
+                                            self.__matches_map_column['relevant']: str, 
+                                            self.__matches_map_column['scores']: str})
+
+        self.__img_paths = pd.read_csv(self._data_folder / img_list_csv_path,
+                                       usecols=[self.__img_list_map_column['img_id'],
+                                                self.__img_list_map_column['image_path']],
+                                       dtype={self.__img_list_map_column['img_id']: int,
+                                              self.__img_list_map_column['image_path']: str})
+
+        (self.__n_relevant,
+         self.__n_queries,
+         self.__index2imgid,
+         self.__relevant_arr,
+         self.__relevance_scores) = self.__parse_match_csv()
+
+        self._imgid2paths = dict(zip(self.__img_paths[self.__img_list_map_column['img_id']],
```

Comment:
Since there is only one path for each image id, you can rename the dict to `self._imgid2path` for consistency